### DISCUSSION TITLE
Trying to add custom operator similar to notNull

### DISCUSSION THREAD

**Discussion by Mvin** [react-querybuilder/react-querybuilder#220] [Q&A] [answered]
I am looking to add two new operators for 'is true' and 'is false' where like 'notNull' and 'null' the value field is hidden. 

How would I go about doing this? It seems like those two cases are specifically somewhere in the codebase.

**Comment by jakeboone02** [comment, marked as answer]
There are a couple of ways to do it. The [demo code](https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/demo/src/constants.ts#L148-L154) shows how to create a field with `"is"` as the only operator and a checkbox as the value component (see `valueEditorType: 'checkbox'`). In that case, checked = `true` and unchecked = `false`. This method uses the default `ValueEditor` component so there's minimal configuration necessary.

The other way is to do it as you describe, with the operators `"is true"` and `"is false"` but no value editor. That way is a little more complicated. You'll need to set up a custom `ValueEditor` component that returns `null` for those specific operators. If you're using the `formatQuery` method, you'll also need to set up a custom `valueProcessor` function to take care of the special case.

I've set up [this codesandbox](https://codesandbox.io/s/react-querybuilder-is-true-is-false-operators-ibrpv) to demonstrate both techniques. Field 1 is configured to use the default components, Field 2 uses custom `ValueEditor`/`valueProcessor`, and Field 3 is just a basic text input so you can see how to incorporate the default `ValueEditor` component into your custom component.

**Reply by Mvin** [reply]
Thanks for the quick response, ended up going with a custom value editor for our use case.
